spynnaker/pyNN/models/neural_projections/connectors/kernel_connector.py

```diff
@@ -0,0 +1,442 @@
+            all_delays)
+        self._all_pre_in_range_weights[pre_vs][post_vs] = numpy.array(
+            all_weights)
+
+    # Get the number of connections
+    def get_num_conns(
+            self, weights, delays, pre_vertex_slice, post_vertex_slice):
+        if (str(pre_vertex_slice) not in self._num_conns or
+                str(post_vertex_slice) not in
+                self._num_conns[str(pre_vertex_slice)]):
+            self.compute_statistics(weights, delays,
+                                    pre_vertex_slice, post_vertex_slice)
+
+        return self._num_conns[str(pre_vertex_slice)][str(post_vertex_slice)]
+
+    @overrides(AbstractConnector.get_delay_maximum)
+    def get_delay_maximum(self, delays):
+        # I think this is overestimated, but not by much
+        n_conns = (
+            self._pre_w * self._pre_h * self._kernel_w * self._kernel_h)
+        # Use the kernel delays if user has supplied them
+        if self._krn_delays is not None:
+            return self._get_delay_maximum(self._krn_delays, n_conns)
+
+        # if not then use the values that came in
+        return self._get_delay_maximum(delays, n_conns)
+
+    @overrides(AbstractConnector.get_n_connections_from_pre_vertex_maximum)
+    def get_n_connections_from_pre_vertex_maximum(
+            self, delays, post_vertex_slice, min_delay=None, max_delay=None):
+        # This is clearly a cop-out, but it works at the moment:
+        # I haven't been able to make this break for "standard usage"
+        return numpy.clip(
+            self._kernel_h * self._kernel_w * self._pre_h * self._pre_w,
+            0, 255)
+
+    @overrides(AbstractConnector.get_n_connections_to_post_vertex_maximum)
+    def get_n_connections_to_post_vertex_maximum(self):
+        # Again as above this is something of a cop-out and we can
+        # probably do better
+        return numpy.clip(
+            self._kernel_h * self._kernel_w * self._post_h * self._post_w,
+            0, 255)
+
+    @overrides(AbstractConnector.get_weight_maximum)
+    def get_weight_maximum(self, weights):
+        # I think this is overestimated, but not by much
+        n_conns = (
+            self._pre_w * self._pre_h * self._kernel_w * self._kernel_h)
+        # Use the kernel delays if user has supplied them
+        if self._krn_weights is not None:
+            return self._get_delay_maximum(self._krn_weights, n_conns)
+
+        return self._get_weight_maximum(weights, n_conns)
+
+    def __repr__(self):
+        return "KernelConnector(shape_kernel[{},{}])".format(
+            self._kernel_w, self._kernel_h)
+
+    @overrides(AbstractConnector.create_synaptic_block)
+    def create_synaptic_block(
+            self, weights, delays, pre_slices, pre_slice_index, post_slices,
+            post_slice_index, pre_vertex_slice, post_vertex_slice,
+            synapse_type):
+        n_connections = self.get_num_conns(weights, delays,
```

Comment:
I'm not sure it's that straightforward as the storage is done on a slice-by-slice basis, but I'll see if I can work this out.  (This is also only ever called from create_synaptic_block(...) as well, so it would only ever be used if we switched to using this connector on-host rather than on-chip...)

Reply:
Turns out it was possible and quite straightforward to implement.  I've tested using this connector on-host and it passes the relevant tests (as does the on-chip version).